sdk/couchbase-core/src/httpx/base.rs

```diff
@@ -0,0 +1,97 @@
+use async_trait::async_trait;
+use bytes::Bytes;
+use futures::{Stream, StreamExt};
+use http::{StatusCode};
+use serde::de::DeserializeOwned;
+use typed_builder::TypedBuilder;
+use crate::httpx::error::ErrorKind::Generic;
+use crate::httpx::error::{Result as HttpxResult};
+
+#[derive(Debug)]
+pub struct Request {
+    pub user_agent: String,
+    pub uri: String,
+    pub basic_auth_user: String,
+    pub basic_auth_pass: String,
+    pub method: http::Method,
+    pub content_type: String,
+    pub on_behalf_of: Option<OnBehalfOfInfo>,
+    pub body: Option<String>,
+}
+
+#[derive(PartialEq, Eq, Debug, TypedBuilder)]
```

Comment:
All of the request type structs probably need to have #[non_exhaustive] on them.